### PR TITLE
[iOS] expose keyCode in onKeyPress event (TextInput)

### PR DESCRIPTION
**Pull request by robertherber**
## Summary

This exposes the ios keyCode to onKeyPress events. This makes it possible to support richer keyboard interactions, like in [react-native-hotkeys](https://github.com/Kingstinct/react-native-hotkeys/blob/main/src/utils/mapIosKeyCode.ts) which provides a uniform interface for both web and iOS.

## Changelog

[IOS] [ADDED] - Added iosKeyCode to onKeyPress events for iOS 13.4+

## Test Plan

This is used live in conjunction (with patch-package) with react-native-hotkeys in https://www.schedulist.app/. All it does is expose an additional property containing the iOS key code to onKeyPress events.

### CODE REVIEW

**Comment by robertherber** [comment]
Why is the CircleCI Pipeline failing? Seems to fail at start..

File: Libraries/Text/TextInput/RCTBaseTextInputView.m [hunk 1]
```diff
@@ -360,15 +361,39 @@ - (void)textInputDidEndEditing
...
+- (void)pressesBegan:(NSSet<UIPress *> *)presses withEvent:(UIPressesEvent *)event{
+  UIPress* firstPress = [[presses allObjects] objectAtIndex:0];
+  
+  if (@available(iOS 13.4, *)) {
+    if(firstPress != nil){
+      UIKey* key = firstPress.key;
+      CFIndex keyCode = (CFIndex) key.keyCode;
+      NSInteger *iosKeyCode = (NSInteger*) keyCode;
+      
+      if(key != nil){
```
Comment (ryancat) [inline Libraries/Text/TextInput/RCTBaseTextInputView.m:384]: If key is nil, wouldn't `CFIndex keyCode = (CFIndex) key.keyCode;` get exceptions already?

Comment (robertherber) [reply Libraries/Text/TextInput/RCTBaseTextInputView.m:382]: Good point - if key actually happens to be null that is of course the case.

**Comment by robertherber** [comment]
@ryancat Could you please reopen this?

**Pull request closed without merging**